From 4b101ab02ea762f70903e894a4b002b229526a4a Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michaelni@gmx.at>
Date: Mon, 12 Aug 2013 22:52:23 +0200
Subject: [PATCH] avformat/asfdec: call ff_read_frame_flush() in asf_read_pts()

flushing just the asf demuxer but not the cores buffers leads to inconsistencies
and a "random" packet position which later causes an assertion failure

Fixes Ticket2853

Signed-off-by: Michael Niedermayer <michaelni@gmx.at>
---
 libavformat/asfdec.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavformat/asfdec.c b/libavformat/asfdec.c
index cffbcc4..0d6370d 100644
--- a/libavformat/asfdec.c
+++ b/libavformat/asfdec.c
@@ -1414,6 +1414,7 @@ static int64_t asf_read_pts(AVFormatContext *s, int stream_index,
     if (avio_seek(s->pb, pos, SEEK_SET) < 0)
         return AV_NOPTS_VALUE;
 
+    ff_read_frame_flush(s);
     asf_reset_header(s);
     for (;;) {
         if (av_read_frame(s, pkt) < 0) {
-- 
1.8.5.5

